### ISSUE TITLE
Warn about shell expressions inside quotes

### ISSUE THREAD

**Issue by jkroepke** [koalaman/shellcheck#2099]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
mkdir dir
touch dir/file
ls "dir/*" # ls: dir/*: No such file or directory
ls "dir/"* # dir/file
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

It should raise a warning about `ls "dir/*"`. Using star (or other shell expressions) inside quotes won't work.
